addons/survey/models/survey_user.py

```diff
@@ -535,10 +535,20 @@ def create(self, vals_list):
         return super(SurveyUserInputLine, self).create(vals_list)
 
     def write(self, vals):
-        score_vals = self._get_answer_score_values(vals, compute_speed_score=False)
-        if not vals.get('answer_score'):
-            vals.update(score_vals)
-        return super(SurveyUserInputLine, self).write(vals)
+        res = True
+        for line in self:
+            vals_copy = {**vals}
+            getter_params = {
+                'user_input_id': self.user_input_id.id,
+                'answer_type': self.answer_type,
+                'question_id': self.question_id.id,
```

Comment:
```'question_id': self.question_id.id```will fail if self is a record set with several records. Same for ```self.user_input_id``` and for ```self.answer_type```. ```self```must be replaced by ```line```